src/main/java/eu/lundegaard/liferay/db/setup/core/SetupFragments.java

```diff
@@ -0,0 +1,252 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Lundegaard a.s.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package eu.lundegaard.liferay.db.setup.core;
+
+import com.liferay.fragment.model.FragmentEntry;
+import com.liferay.fragment.service.FragmentCollectionLocalServiceUtil;
+import com.liferay.fragment.service.FragmentEntryLinkLocalServiceUtil;
+import com.liferay.fragment.service.FragmentEntryLocalServiceUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.service.ServiceContext;
+import eu.lundegaard.liferay.db.setup.domain.Fragment;
+import eu.lundegaard.liferay.db.setup.domain.FragmentCollection;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * @author michal.volf@lundegaard.eu
+ */
+public class SetupFragments {
+
+    private static final Log LOG = LogFactoryUtil.getLog(SetupFragments.class);
+
+    private SetupFragments() {}
+
+    public static void setupFragments(List<FragmentCollection> fragmentCollections, long userId, long groupId) {
+        for (FragmentCollection fragmentCollection : fragmentCollections) {
+            ServiceContext serviceContext = new ServiceContext();
+
+            switch (fragmentCollection.getSetupAction()) {
+                case "create":
+                    createFragmentCollection(fragmentCollection, userId, groupId, serviceContext);
+                    break;
+                case "update":
+                    updateFragmentCollection(fragmentCollection, groupId);
+                    break;
+                case "delete":
+                    deleteFragmentCollection(fragmentCollection, groupId);
+                    break;
+                default:
+                    throw new IllegalStateException("Illegal setup action " + fragmentCollection.getSetupAction());
```

Comment:
IllegalStateException = component is in wrong state, you want IllegalArgumentException